### PR TITLE
Features: Set USERNAME to _REMOTE_USER if available

### PR DESCRIPTION
**Pull request by samruddhikhandale**
Helps fix https://github.com/devcontainers/features/issues/307

### CODE REVIEW

File: src/anaconda/install.sh [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 
 VERSION=${VERSION:-"latest"}
-USERNAME=${USERNAME:-"automatic"}
+USERNAME="${USERNAME:-"${_REMOTE_USER:-"automatic"}"}"
```
Comment (samruddhikhandale) [inline src/anaconda/install.sh:12]:
```suggestion
USERNAME="${USERNAME:-"${_REMOTE_USER:-"_CONTAINER_USER:-"automatic""}"}"
```

Not sure if we should also consider `_CONTAINER_USER` 💭

Comment (jkeech) [reply src/anaconda/install.sh:12]: According to the spec, we shouldn't need to. https://github.com/devcontainers/spec/blob/main/proposals/features-user-env-variables.md

>If no remoteUser is configured, _REMOTE_USER is set to the same value as _CONTAINER_USER.

**Review by jkeech — COMMENTED** [review]
Can you please add some test cases that verify the features work properly for non-standard `remoteUser`s?

**Comment by samruddhikhandale** [comment]
> Can you please add some test cases that verify the features work properly for non-standard `remoteUser`s?

For node, I tired adding a test-scenario as follows 

```
    "test_non_root_user": {
        "image": "ubuntu:focal",
        "features": {
            "common-utils": {
                "username": "app"
            },
            "node": {
                "version": "19.1.0"
            }
        }
    }
```

However, even before the changes in the PR, the tests pass. The `nvm` folder is set with correct user permissions & node is installed fine.

image

I thought of repro-ing the scenario using `dockerfile` which creates this user, however the features test command doesn't support that. Need to fix that. // cc - @joshspicer 

image

@jkeech Not sure how else I could write tests 😕

**Comment by samruddhikhandale** [comment]
Closing this in favor of https://github.com/devcontainers/features/pull/264

**Pull request closed without merging**